src/hooks/useSendFlowQrCodeScanner.tsx

```diff
@@ -0,0 +1,194 @@
+  usePopToTop: false;
+}
+
+interface QRCodeScreenReturn {
+  handlers: QRCodeScreenHandlers;
+  state: QRCodeScreenState;
+  config: QRCodeScreenConfig;
+  /** Manual input overlay component (not used for send flow) */
+  ManualInputOverlay?: undefined;
+}
+
+/**
+ * Custom hook for Send Flow QR Code Scanner Screen functionality
+ *
+ * This hook provides all the necessary state, handlers, and configuration
+ * specifically for address input in the send flow.
+ *
+ * @returns Object containing handlers, state, and configuration
+ */
+export const useSendFlowQrCodeScanner = (): QRCodeScreenReturn => {
+  const { t } = useAppTranslation();
+  const navigation =
+    useNavigation<NativeStackNavigationProp<RootStackParamList>>();
+
+  const [isProcessingQRScan, setIsProcessingQRScan] = useState(false);
+
+  const { clearQRData } = useQRDataStore();
+
+  const {
+    searchAddress,
+    searchResults,
+    isValidDestination,
+    isSearching,
+    destinationAddress,
+  } = useSendRecipientStore();
+  const { saveRecipientAddress, selectedTokenId } =
+    useTransactionSettingsStore();
+
+  // Configuration for Send Flow
+  const config: QRCodeScreenConfig = {
+    showHeaderRight: false,
+    usePopToTop: false,
+  };
+
+  // State for Send Flow
+  const state: QRCodeScreenState = {
+    manualInput: "",
+    isConnecting: isProcessingQRScan && isSearching,
+    error: "",
+    showManualInput: false,
+    title: t("qrCodeScannerScreen.title"),
+    scannerTitle: t("sendPaymentScreen.scanQRCodeText"),
+    context: QRCodeSource.ADDRESS_INPUT,
+  };
+
+  // Handle closing the screen
+  const handleClose = useCallback(() => {
+    navigation.goBack();
+  }, [navigation]);
+
+  // Handle header left button press
+  const handleHeaderLeft = useCallback(() => {
+    handleClose();
+  }, [handleClose]);
+
+  // Handle QR code scanning
+  const handleQRCodeScanned = useCallback(
+    (data: string) => {
+      if (isValidStellarAddressForQR(data)) {
+        // Start processing the QR scan
+        setIsProcessingQRScan(true);
+        // Start the address search
+        searchAddress(data);
+        // Track analytics and call onRead
+        analytics.trackQRScanSuccess(QRCodeSource.ADDRESS_INPUT);
+      }
+      // If invalid, do nothing - the QR scanner will continue scanning
+    },
+    [searchAddress],
+  );
+
+  // Handle search results when QR scan is processed
+  useEffect(() => {
+    if (
+      isProcessingQRScan &&
+      isValidDestination &&
+      !isSearching &&
+      searchResults.length > 0
+    ) {
+      // Save the recipient address to transaction settings store
+      // The send recipient store already has the correct address from the search
+      saveRecipientAddress(destinationAddress);
+
+      // Navigate directly to transaction amount screen with the selected token
+      // Reset the navigation stack so back action goes to main tab instead of QR scanner
+      navigation.reset({
```

Comment:
this seems like a bit of a brute force, could we use `navigation.pop(n)`, `navigation.popTo(route)` or `navigation.replace(route)` instead to handle this ?